source/slang/slang-ir-util.cpp

```diff
@@ -175,7 +175,9 @@ IRInst* maybeSpecializeWithGeneric(IRBuilder& builder, IRInst* genericToSpecaili
 // Returns true if is not possible to produce side-effect from a value of `dataType`.
 bool isValueType(IRInst* dataType)
 {
-    dataType = getResolvedInstForDecorations(unwrapAttributedType(dataType));
+    if(auto t = as<IRType>(dataType))
```

Comment:
This if is unnecessary.

Reply:
I changed the type of unwrapAttributedType to be on IRType, so it's required now.

Reply:
That means we are checking the opcode twice, once for the dynamic cast and once for the actual unwrapping, which seems unnecessary.

Reply:
We should feel comfortable to accept IRInst where a type is expected, and rely on static cast to convert between IRInst and IRType because there is not a clean way to separate a type and an inst anyways. IRType has no more meaning other than clarifying what an IRInst is for in code, but for unwrapAttributeType, it is verify clear what the param is, so it accepting IRInst is not wrong and I don't see a need for the change.

Reply:
ok, I've reverted that change